dubbo-plugin/dubbo-qos/src/main/java/org/apache/dubbo/qos/server/handler/QosProcessHandler.java

```diff
@@ -93,6 +93,15 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
         }
     }
 
+    @Override
+    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+        ScheduledFuture<?> future = welcomeFuture;
+        if (future != null && !future.isCancelled()) {
```

Comment:
this place need to judge if it's a IdleStateEvent

Reply:
Thank you, it’s my mistake, I forgot to add my judgment.